pallets/asset-registry/src/lib.rs

```diff
@@ -590,6 +590,13 @@ impl<T: Config> ShareTokenRegistry<T::AssetId, Vec<u8>, T::Balance, DispatchErro
 	}
 }
 
+impl<T: Config> BondRegistry<T::AssetId, Vec<u8>, T::Balance, DispatchError> for Pallet<T> {
+	fn create_bond_asset(existential_deposit: T::Balance) -> Result<T::AssetId, DispatchError> {
+		let bounded_name: BoundedVec<u8, T::StringLimit> = Self::to_bounded_name(Vec::<u8>::new())?;
+		Self::register_asset(bounded_name, AssetType::Bond, existential_deposit, None, None)
+	}
+}
+
```

Comment:
i would make this little more generic. 

there is a need to create stableswap share asset too which requires similar change in registry. 

I can do it in separate PR - just this generic create asset trait which takes type as parameter. 

sounds good ?

Reply:
Good idea. I would also consider moving the asset type to `primitives` crate so we don't need to make other pallets dependent on asset registry pallet.